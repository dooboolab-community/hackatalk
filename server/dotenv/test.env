JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
