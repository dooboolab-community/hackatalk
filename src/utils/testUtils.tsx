import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
