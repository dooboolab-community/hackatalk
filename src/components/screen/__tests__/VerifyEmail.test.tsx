import 'react-native';

import React, { ReactElement } from 'react';
import { RenderResult, act, fireEvent, render, wait } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Screen from '../VerifyEmail';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderResult;

// const mockSendVerification = [
//   {
//     request: {
//       query: MUTATION_SEND_VERIFICATION,
//       variables: {
//         email: 'test@email.com',
//       },
//     },
//     result: {
//       data: {
//         sendVerification: true,
//       },
//     },
//   },
// ];

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps({
      route: {
        params: {
          email: 'test@email.com',
        },
      },
    });
    component = createTestElement(
      <Screen {...props} />,
    );
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('Interaction', () => {
  beforeEach(() => {
    props = createTestProps({
      route: {
        params: {
          email: 'test@email.com',
        },
      },
    });
    component = createTestElement(
      <Screen {...props} />,
    );
    testingLib = render(component);
  });

  it('should simulate email button press', async () => {
    const btn = testingLib.queryByTestId('touch-email');
    act(() => {
      fireEvent.press(btn);
    });
    await act(() => wait());
  });

  it('should simulate next button press', () => {
    testingLib.debug();
    const btn = testingLib.queryByTestId('btn-next');
    act(() => {
      fireEvent.press(btn);
    });
    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
