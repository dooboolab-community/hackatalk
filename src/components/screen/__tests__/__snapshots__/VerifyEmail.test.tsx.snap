// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction should simulate next button press 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "lineHeight": 28,
              "textAlign": "center",
            },
            Object {
              "marginBottom": 24,
            },
          ]
        }
      >
        Verification link has been sent. Please check your email. If you have not received your email please check out spam inbox.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1E6EFA",
              "fontSize": 24,
            },
          ]
        }
      >
        test@email.com
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "marginTop": 24,
          }
        }
        testID="touch-email"
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "lineHeight": 28,
                "textAlign": "center",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Resend email
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        delayPressIn={30}
        style={
          Object {
            "marginTop": 80,
            "paddingHorizontal": 20,
            "width": "100%",
          }
        }
        testID="btn-next"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "blue",
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#1E6EFA",
                "borderWidth": 0,
                "height": 48,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#069ccd",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                },
              ]
            }
          >
            Return to sign in
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`Rendering renders without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "lineHeight": 28,
              "textAlign": "center",
            },
            Object {
              "marginBottom": 24,
            },
          ]
        }
      >
        Verification link has been sent. Please check your email. If you have not received your email please check out spam inbox.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1E6EFA",
              "fontSize": 24,
            },
          ]
        }
      >
        test@email.com
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "marginTop": 24,
          }
        }
        testID="touch-email"
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "lineHeight": 28,
                "textAlign": "center",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Resend email
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        delayPressIn={30}
        style={
          Object {
            "marginTop": 80,
            "paddingHorizontal": 20,
            "width": "100%",
          }
        }
        testID="btn-next"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "blue",
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#1E6EFA",
                "borderWidth": 0,
                "height": 48,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#069ccd",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                },
              ]
            }
          >
            Return to sign in
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
